Simplenote/SPAppDelegate.m

```diff
@@ -295,6 +295,25 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     return YES;
 }
 
+- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray * _Nullable))restorationHandler {
+    
+    if (userActivity.userInfo != nil) {
+        if (userActivity.userInfo[CSSearchableItemActivityIdentifier] != nil) {
+            NSString *uniqueIdentifier = userActivity.userInfo[CSSearchableItemActivityIdentifier];
```

Comment:
Suggestion: Since in ObjC messages to `nil` are allowed, perhaps something like this would be easier to read:

```
NSString *uniqueIdentifier = userActivity.userInfo[CSSearchableItemActivityIdentifier];
if (uniqueIdentifier == nil) {
   return false;
}

SPBucket *noteBucket = [_simperium bucketForName:@"Note"];
Note *note = [noteBucket objectForKey:uniqueIdentifier];
if (note == nil) {
   return false;
}

(...)

return true;
```

(Reason: less indentation is way easier to understand, plus, this way we'd tell the OS if a Note is no longer available).